lib/sarif/gosec_sarif.rb

```diff
@@ -1,32 +1,46 @@
+require 'salus/bugsnag'
+
 module Sarif
   class GosecSarif < BaseSarif
+    include Salus::SalusBugsnag
+
     GOSEC_URI = 'https://github.com/securego/gosec'.freeze
 
     def initialize(scan_report)
       super(scan_report)
       @uri = GOSEC_URI
-      begin
-        json_obj = JSON.parse(scan_report.log(''))
-        issues = json_obj['Issues']
-        errors = json_obj['Golang errors']
-        parsed_errors = []
-        errors.each do |key|
-          key[1].each do |location|
-            location['uri'] = key[0]
-            parsed_errors << parse_error(location)
-          end
+      @issues = Set.new
+      @logs = parse_scan_report!(scan_report)
+    end
+
+    def parse_scan_report!(scan_report)
+      json_obj = JSON.parse(scan_report.log(''))
+      issues = json_obj['Issues']
+      errors = json_obj['Golang errors']
+      parsed_errors = []
+      errors.each do |key|
+        key[1].each do |location|
+          location['uri'] = key[0]
+          parsed_errors << parse_error(location)
         end
-        @logs = issues.concat parsed_errors
-      rescue JSON::ParserError
-        @logs = []
       end
+      parsed_errors.compact!
+      issues.concat parsed_errors
+    rescue JSON::ParserError => e
+      bugsnag_notify(e.message)
+      []
     end
 
     def parse_error(error)
       line = error['line'].to_i
       column = error['column'].to_i
       line = 1 if line.zero?
       column = 1 if column.zero?
+
+      id = error['error'] + error['uri'] + line.to_s
```

Comment:
I added a comment about how the `id = ...` could be buggy.   The comment was marked as resolved, but I don't see  any changes to the line?